src/gtk/view_dialog.c

```diff
@@ -44,8 +44,20 @@ do_view_or_edit_file (gftp_window_data * fromwdata, int is_view)
   if (S_ISDIR (curfle->st_mode))
     {
       if (is_view)
-        ftp_log (gftp_logging_error, NULL,
-                 _("View: %s is a directory. Cannot view it.\n"), curfle->file);
+      {
+        if (strcmp (gftp_protocols[fromwdata->request->protonum].name, "Local") == 0)
```

Comment:
This string comparison is fine since that's how the code is currently structured. If you are looking for other improvements to gftp, then I suggest adding to the supported_gftp_protocols structure a new enum field for each of the supported protocols. Right now, if there's a slight typo in the name then the compiler then the compiler won't break the build.